app/src/main/java/com/base/firebase_operations/MainActivity2.kt

```diff
@@ -0,0 +1,66 @@
+package com.base.firebase_operations
+
+import android.content.Intent
+import android.net.Uri
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Button
+import android.widget.ImageView
+import android.widget.Toast
+import com.google.firebase.ktx.Firebase
+import com.google.firebase.storage.ktx.storage
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.tasks.await
+import kotlinx.coroutines.withContext
+
+class MainActivity2 : AppCompatActivity() {
+
+    private val Requestcode=0
+    private var currentfile: Uri?=null
+    lateinit var imageholder:ImageView
+    val imagereff=Firebase.storage.reference
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main2)
+        imageholder=findViewById(R.id.viewimage)
+        val upload: Button=findViewById(R.id.button5)
+        imageholder.setOnClickListener{
+            Intent(Intent.ACTION_GET_CONTENT).also {
+                it.type="image/*"
+                startActivityForResult(it,Requestcode)
+            }
+        }
+        upload.setOnClickListener {
+            UploadImagetoCloud("TestImages")
+        }
+    }
+
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
```

Comment:
Here the onActivityResult method is called when an activity is launched with startActivityforResult is finished and it returns a result. It generally means starting an activity which would be returning a result when it is finished.
The we just used a general check for checking that is the Result successfull and the request code is the one or the same as it was used while starting the activity and if both condition are true then we executed the code inside the if block which here is getting the url of the selected image using data intent and passing it to the currentfile variable.